File: src/main/java/pl/khuzzuk/wfrp/helper/service/knowledge/CurrentMagicKnowledge.java
package pl.khuzzuk.wfrp.helper.service.knowledge;

import lombok.Getter;
import lombok.Setter;
import pl.javahello.DTO;
import pl.khuzzuk.wfrp.helper.model.knowledge.Skill;
import pl.khuzzuk.wfrp.helper.model.knowledge.magic.SpellSchoolLevel;
import pl.khuzzuk.wfrp.helper.model.world.Realm;

import javax.validation.constraints.NotNull;
import java.util.List;

@DTO
@Getter
@Setter
public class CurrentMagicKnowledge {
    private List<SpellSchoolLevel> currentSpellSchools;
    private @NotNull Realm realm;
    private List<Skill> currentSkills;
}

File: src/wfrp-helper-web/src/model/knowledge/CurrentMagicKnowledge.js
import Realm from "../realm/Realm";
import SpellSchoolLevel from "./SpellSchoolLevel";
import Skill from "./Skill";

export default class CurrentMagicKnowledge {
    currentSpellSchools: SpellSchoolLevel[] = [];
    currentSkills: Skill[] = [];
    realm: Realm;
}